### DISCUSSION TITLE
colorspace for grayscale resizing

### DISCUSSION THREAD

**Discussion by exekutive** [ImageMagick/ImageMagick#7448] [Help] [answered]
### ImageMagick version

7.1.1

### Operating system, version and so on

any

### Description

So in IM documentation: [resize](https://imagemagick.org/Usage/resize/#resize_colorspace), [color mods](https://imagemagick.org/Usage/color_mods/#sigmoidal), and also in @nrobidoux [treatment](https://imagemagick.org/Usage/filter/nicolas/#upsampling) of resize, it is proven that enlarging operations should not be performed in a sRGB (or gamma compensated) colorspace. It is better to resize in a linear colorspace or use sigmoidal contrast.

for example:
```
magick {input} -colorspace RGB +sigmoidal-contrast 7.5 \
          -filter Lanczos -define filter:blur=.9891028367558475 \
          -distort Resize 500% \
          -sigmoidal-contrast 7.5 -colorspace sRGB {output}
```
My question is what if you are working with an image that was converted using the `-colorspace gray` operator? Is this a linear colorspace or no? Should I perform a linear colorspace or sigmoidal contrast conversion before the resize, and then back to `gray` after?

**Comment by fmw42** [comment, marked as answer]
That is a non-linear grayscale conversion. If you want a linear-gray use -colorspace LinearGray.  See the list of colorspaces (magick -list colorspace)

**Comment by nrobidoux** [comment]
If you are enlarging or at least not downsampling much, using a well-chosen nonlinear colorspace is often a good thing, sigmoidal or pure gamma (2, say) RGB, or even Lab being good choices.

**Reply by exekutive** [reply]
enlarging 300%
Here's what I'll be testing ...

```
    magick mogrify -colorspace LinearGray +sigmoidal-contrast 11.5 \
      -filter Lanczos -define filter:blur=.9891028367558475 \
      -distort Resize 300% \
      -sigmoidal-contrast 11.5 -colorspace gray $work

```
